pkg/templates/loader/loader.go

```diff
@@ -8,15 +8,31 @@ import (
 
 // SchemaLoader define the interface for loading schema from template.
 type SchemaLoader interface {
-	Load(rootDir, templateName, templateVersion string) (*types.Schema, error)
+	Load(rootDir, templateName string, mode Mode) (*types.Schema, error)
```

Comment:
Whether it's from schema file or not is the implementation detail of SchemaLoader. Why do we expose it to the caller

Reply:
Expose to support different modes.
- Load from file: mainly used in walrus server. While walrus sync template schema, when there is a schema.yaml, will use the schema user defined in schema.yaml directly.
- Merge: mainly used in CLI, while user already defined a customized schema.yaml, and then updated the varaibles.tf this mode will regenerate the schema from variables.tf and merge the result with the schema from schema.yaml, this mode will include both new generated and user customized.
- Original: only return the generated, ignore the customized.